app/models/pawn.rb

```diff
@@ -1,2 +1,55 @@
 class Pawn < Piece
+  def valid_move?(destination_x, destination_y)
+    return false unless destination_x.between?(0, 7) && destination_y.between?(0, 7)
+    # return false if is_obstructed?(destination_x, destination_y)
+    # include the above line once that method is added to the piece model
+
+    possible_moves = []
+
+    if color == 'white'
+      possible_moves << [position_x.to_i, position_y.to_i + 1]
+      if position_y = 1
+        possible_moves << [position_x.to_i, position_y.to_i + 2]
+      end
+    elsif color == 'black'
+      possible_moves << [position_x.to_i, position_y.to_i - 1]
+      if position_y = 6
+        possible_moves << [position_x.to_i, position_y.to_i - 2]
+      end
+    end
+
+    destination_piece = Piece.find_by(position_x: destination_x, position_y: destination_y)
```

Comment:
For this method. No need to worry about if there is a piece in the destination square. We can check for that in another method. (Makes things easier on you :smile:)

Eventually, we'll have another method called `move_to!` which will work something like this.

```Ruby
#pseudo code :)
if valid_move?
  return remove_opposing_piece if opposing_piece_in_destination?
  return error if same_color_in_destination?
  move_current_piece
end
```

Reply:
@mscoutermarsh Okay, should I also remove the RSpec tests where there's a piece in the destination?

Reply:
Up to you. 😄